src/js/controllers/ctrl-selectedwidget.js

```diff
@@ -54,21 +54,28 @@ app.controller('SelectedWidgetController', function(
 		})
 	}
 
-	const _editWidget = () => {
-		if ($scope.selected.editable) {
-			Materia.Coms.Json.send('widget_instance_lock', [$scope.selected.widget.id]).then(success => {
-				if (success) {
-					if ($scope.selected.widget.is_draft) {
-						window.location = $scope.selected.edit
-					} else {
-						$scope.show.editPublishedWarning = true
-					}
+	const _editWidgetPromise = () => {
+		return Materia.Coms.Json.send('widget_instance_edit_perms_verify', [
+			$scope.selected.widget.id
+		]).then(response => {
+			if (response.is_locked) {
+				$scope.alert.msg =
+					'This widget is currently locked, you will be able to edit this widget when it is no longer being edited by somebody else.'
```

Comment:
I think we probably shouldn't use `locked` with the user
```suggestion
					'Someone else is editing this widget, you will be able to edit after they finish.'
```

Reply:
Feels nicer. I'll swap it in.